Add version to .htaccess

LukasReschke opened the pull request:
Currently if a user does not replace the .htaccess file with the new update this can lead to serious problems in case Apache is used as webserver.

This commit adds the version to the .htaccess file and the update routine fails in case not the newest version is specified in there. This obviously means that every release has to update the version specified in .htaccess as well. But I see no better solution for it.

Fixes https://github.com/owncloud/core/issues/9937 and a lot of other bug reports.

Testplan:
- Update the version in version.php and trigger the update via web => Should fail
- Update the version in version.php, adjust the .htaccess as well and trigger the update vie web => Should work

Known bug that needs to be addressed somewhere else: `occ` does not run the pre-update checks while the web interface does. Something for another PR. Not here.

@karlitschek I know that this is additional pain when creating releases. However, I see no other proper solution for this. Checking for the bad rewrites seems like a very bad solution and will give us trouble in future as well.

MorrisJobke commented:
@LukasReschke Do you want to get this back in a working state or should I?

LukasReschke commented:
I'll take care of this one then.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/4e97998e-3516-4d4c-81ac-ede4b53374fd) was created.

LukasReschke commented:
I rebased and also added some unit-tests to ensure that CI will fail in case the `.htaccess` version has not been updated.

Though this requires that @karlitschek uses PRs for version bumps or we will have a broken master if the update has been forgotten ;-)

MorrisJobke commented:
@LukasReschke Can you add this to the install assistent? Otherwise the error is just shown once you send the install form.

LukasReschke commented:
> @LukasReschke Can you add this to the install assistent? Otherwise the error is just shown once you send the install form.

I don't really want to mutate that zombie in the install assistant even more. At initial installation the chance that the correct `.htaccess` is there is actually pretty high ;-)

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/4e97998e-3516-4d4c-81ac-ede4b53374fd) completed: **1 new issues, 3 updated code elements**

MorrisJobke commented:
@LukasReschke Okay...It works nice for the upgrade process and shows a warning after the installation. I'm fine with that :+1:

LukasReschke commented:
Do I really have to create a new PR now for CI or is there any other magic that can help…?

MorrisJobke commented:
@LukasReschke I think you need a new PR

LukasReschke commented:
https://github.com/owncloud/core/pull/13172
